examples/transient_adiabatic_ce/pin-td.py

```diff
@@ -0,0 +1,128 @@
+import numpy as np
+import openmc
+import openmc.mgxs
+import openmc.kinetics as kinetics
+
+from geometry_pin import geometry
+materials_file = openmc.Materials(geometry.get_all_materials().values())
+
+###############################################################################
+# Define problem settings
+
+settings_file = openmc.Settings()
+settings_file.batches = 200
+settings_file.inactive = 100
+settings_file.particles = 1000
+settings_file.output = {'tallies': False}
+
+# Create an initial uniform spatial source distribution over fissionable zones
+lower_left = [-0.62992, -0.62992, -182.88]
+upper_right = [0.62992, 0.62992, 182.88]
+uniform_dist = openmc.stats.Box(lower_left, upper_right, only_fissionable=True)
+settings_file.source = openmc.source.Source(space=uniform_dist)
+
+sourcepoint = dict()
+sourcepoint['batches'] = [settings_file.batches]
+sourcepoint['separate'] = True
+sourcepoint['write'] = True
+settings_file.sourcepoint = sourcepoint
+
+# For source convergence checks, add a mesh that can be used to calculate the
+# Shannon entropy
+entropy_mesh = openmc.RegularMesh()
+entropy_mesh.lower_left = lower_left
+entropy_mesh.upper_right = upper_right
+entropy_mesh.dimension = [1,1,30]
+settings_file.entropy_mesh = entropy_mesh
+
+###############################################################################
+# Define transient problem parameters
+
+# Create pin cell mesh
+full_pin_cell_mesh = openmc.RegularMesh()
+full_pin_cell_mesh.type = 'regular'
+full_pin_cell_mesh.dimension = [1,1,30]
+full_pin_cell_mesh.lower_left = [-0.62992,-0.62992,-182.88]
+full_pin_cell_mesh.width =[0.62992*2,0.62992*2,12.192]
+
+# Instantiate an EnergyGroups object for the diffuion coefficients
+fine_groups = openmc.mgxs.EnergyGroups()
+fine_groups.group_edges = [0., 0.13, 0.63, 4.1, 55.6, 9.2e3, 1.36e6, 1.0e7]
+
+# Instantiate an EnergyGroups object for the transient solve
+energy_groups = openmc.mgxs.EnergyGroups()
+energy_groups.group_edges = [0., 0.13, 0.63, 4.1, 55.6, 9.2e3, 1.36e6, 1.0e7]
+
+# Instantiate an EnergyGroups object for one group data
+one_group = openmc.mgxs.EnergyGroups()
+one_group.group_edges = [fine_groups.group_edges[0], fine_groups.group_edges[-1]]
+
+# Instantiate a clock object
+t_outer = np.arange(0., 1.0, 5.e-1)
+clock = openmc.kinetics.Clock(start=0., end=0.5, dt_inner=1.e-2, t_outer=t_outer)
+
+# Prescribe the transient as a dictionary of densities and temperatures 
+Transient = {}
+
+# Create entries for each material 
+for material in materials_file:
```

Comment:
Thanks for this! I felt like the dictionary was way too clunky, and I needed some new ideas. I'll play around with what you suggested!

Reply:
I've been looking into this more. Sam originally had made changes to cell.py, material.py, and geometry.py to add a time variable. Then, cell.translation and material.density could be input as arrays along with a time array. Then, when defining your material density, instead of specifying a float, you could give an array. This sounds sort of similar to what you suggested. 
However, all this is for a prescribed transient, which is not really a long-term solution for transients in general. We want to work toward coupled transients eventually. So I'm hesitant to make any time-dependent change in the OpenMC material and cell code. In the end, I think I prefer the input-file-only method of specifying the transient.

Reply:
Makes sense to me!

Reply:
I still want to make it less clunky. I'm thinking about including this dictionary, or a different structure into the kinetics code so that the user can just update parts of it instead of having to generate & fill the whole "transient" for each problem. A work in progress.

Reply:
Nice :D going to be exciting to see it all come together!